Sequencing monitoring

RyandenOtter opened the pull request:
Here are the updates for the monitoring views.

Commit:
feat: tf gcs-bq-ingest module sets bq permissions

... on multiple projects

fixup!

Commit:
fixup roles

Commit:
fixup dockerfil ci check

Commit:
docs: add note on unicode delimiters

Commit:
fix: external configs not found in parent dirs

Commit:
fix: update nested values in configs

Commit:
feat: bq project env-var

Add an environment variable to support overriding the default project
for the BigQuery Client. By default this will be the project in which
the cloud function is deployed.

Commit:
Merge branch 'feat/bq-project-env-var'

Commit:
Merge branch 'fix/configDiscovery'

Commit:
Merge branch 'docs/delimiters'

Commit:
Merge branch 'tf/multi-project'

Commit:
feat: sequencing with backlog publisher / subscriber

* Restructures code into constants and exception modules
* Implements Backlog Publisher / Subscriber algorithm for ordering incrementals
* Implements basic integration tests for Publisher / Subscriber

Commit:
Move utility methods into a utils module
Change the tests to use new utils module.

Commit:
Fix sorting issues

Commit:
Move out constants into their own file
Change import pattern

Commit:
fixup! pylint

Commit:
fixup! fixup! gcb pylint issue

Commit:
Merge pull request #1 from jaketf/feat/move-utils

Move utility methods into a utils module

Commit:
fixup! mypy pylint

Commit:
fixup! flake8

Commit:
fixup! mypy tests

Commit:
support _config/*.sql for bq tranform sql

Commit:
improve performance of wait_on_bq_job

Commit:
wip

Commit:
fixup! handle race condition

Commit:
ordering docs and isort single line rule

Commit:
docs

Commit:
fixup linters

Commit:
fixup import style

Commit:
typing isort single line exclusion

Commit:
fixup gcb no-name-in-module bug

Commit:
add test of subscriber after subscriber exit

Commit:
chores: tf updates, larger machine type, etc.

Commit:
terraform fmt

Commit:
handle abandoned _BACKFILL and other review feedback

Commit:
improve tests

Commit:
fix: handle long running bq jobs

Commit:
chore: add e2e test, fixup terraform

Commit:
ignore pylint redherring import errors

Commit:
fixup! e2e tf to support builds where short_sha is set to empty string.

Commit:
fix TF_VAR env var

Commit:
enable resource manager api

Commit:
enable cloud functions api...

Commit:
add unit test timeout

Commit:
explicit local backend

Commit:
debug missing state file

Commit:
debug

Commit:
relative state path

Commit:
typo .[tf]state

Commit:
fixup docs

Commit:
chore: clean up subscriber

Commit:
fix: don't try to regex match _backlog/* items

Commit:
don't regex match in triage if ordering enabled (this happens later)

Commit:
fix: subscriber monitor get table prefix

Commit:
fix: get_table_prefix issues w/ backlog, backfill and historydone

Commit:
fix: look_for_config_in_parents should return empty string for empty file

Commit:
fix table prefix w/ trailing slash

Commit:
use get_table_prefix instead of removesuffix

Commit:
chore: refactor terraform into pytest fixture to always clean up

Commit:
fix don't removesuffix for start backfill file

Commit:
fixup isort

Commit:
more logging statements fail on untriageable event

Commit:
fix pylint

Commit:
Merge branch 'master' into sequencing-develop

Commit:
feat: env-var t numDmlRowsAffected = 0 as a failure

Commit:
[skip ci] add comment to cloudbuild.yaml

Commit:
[skip ci] update comment in cloudbuild.yaml

Commit:
chore: clean up unused fixture, init files

Commit:
chore: improve terraform printint in pytest fixture

Commit:
better bq job ids

Commit:
fixup regex escaping

Commit:
make pylint happy

Commit:
[skip ci] more docs

Commit:
fix default load config return type

Commit:
fix: fail on failure of children jobs

During multi-statement BQ jobs, child jobs are submitted.
If any of these fail we should consider the job a failure.

Commit:
chore: add test for child job failing behavior

Commit:
fixup flake8

Commit:
fixup flake8

Commit:
fixup flake8

RyandenOtter committed:
There was a missing comma in the example SQL given.

Adding the views and tables for ingest monitoring

Remove unneeded sql

Update README.md

Fix formatting issues on the tables.

Commit:
feat: separate bq storage and compute project env vars

jaketf commented:
you need to sign the first two commits.
```bash
git commit --amend --author="Author Name <email@address.com>"
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
```diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+
+## Monitoring
+### monitored_tables
+To monitor the status of the tables loaded by the cloud function, you must first add records to the monitored_tables table. The records in this table are used by the gcf_ingest_log and gcf_ingest_latest_by_table views to know which tables to report on.
```
```suggestion
To monitor the status of the tables loaded by the Cloud Function, you must first add records to the `monitored_tables` table. The records in this table are used by the `gcf_ingest_log` and` gcf_ingest_latest_by_table` views to know which tables to report on.
```

jaketf replied:
Please provide schema / exmaple rows of the monitored tables table and how you might populate it.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
````diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+```
+
+## Monitoring
+### monitored_tables
````
```suggestion
### `monitored_tables`
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
```diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+### monitored_tables
+To monitor the status of the tables loaded by the cloud function, you must first add records to the monitored_tables table. The records in this table are used by the gcf_ingest_log and gcf_ingest_latest_by_table views to know which tables to report on.
+
+### gcf_ingest_log
```
```suggestion
### `gcf_ingest_log`
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
```diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+To monitor the status of the tables loaded by the cloud function, you must first add records to the monitored_tables table. The records in this table are used by the gcf_ingest_log and gcf_ingest_latest_by_table views to know which tables to report on.
+
+### gcf_ingest_log
+This gives you a view of all of the attempts to load a given chunk. For example, if we wanted to see the entire history of chunk attempts for a 'table_a', we would run the follwing query:
```
chunk is customer specific terminology (and confusing). let's just use the word GCS prefix instead.

jaketf replied:
```suggestion
This gives you a view of all of the attempts to load a given chunk. For example, if we wanted to see the entire history of chunk attempts for a `'table_a'`, we would run the follwing query:
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
````diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+```
+This will give you something like:
+
+|---------------|------------|----------|----------|------------------------------------------------|--------------------------------------------------------------------------------------------- -------|---------------------|---------------------|-------|-----------------------|---------------|--------------------|-----------|------------------------|
````
this might be better as json.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
```diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+| my-project-id | dataset_a  | table_a  | LOAD     | db-dataset_a-table_a-incremental-1900-01-01-08 | gcf-ingest-db-dataset_a-table_a-incremental-1900-01-01-08-_DONEc0f17aa2-740f-4cae-80b7-02c185e5057a | 2021-01-11 20:04:12 | 2021-01-11 20:04:25 | NULL  |    1260021.5833333333 |         14620 | 1.1520882584712373 | SUCCEEDED |                   NULL |
+
+
+To deploy the gcf_ingest_log, you must replace the 'REPLACEME-audit-log-project-id' and 'REPLACEME-audit-log-dataset' with the correct project_id and dataset. You can also replace the `default-dataset` with your preferred dataset.
```
```suggestion
To deploy the `gcf_ingest_log`, you must replace the `'REPLACEME-audit-log-project-id'` and `'REPLACEME-audit-log-dataset'` with the correct project ID and dataset. You can also replace the `default-dataset` with your preferred dataset.
```
It'd be ideal if you added a small bash snippet of how to perform these replacements from the command line.

jaketf replied:
Add a link to how to set up audit log sink to BQ.
https://cloud.google.com/bigquery/docs/reference/auditlogs#defining_a_bigquery_log_sink_using_gcloud

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
```diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+
+To deploy the gcf_ingest_log, you must replace the 'REPLACEME-audit-log-project-id' and 'REPLACEME-audit-log-dataset' with the correct project_id and dataset. You can also replace the `default-dataset` with your preferred dataset.
+
+### gcf_ingest_latest_by_table
```
```suggestion
### `gcf_ingest_latest_by_table`
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
```diff
@@ -59,7 +59,82 @@ DEFAULT_DESTINATION_REGEX = (
+To deploy the gcf_ingest_log, you must replace the 'REPLACEME-audit-log-project-id' and 'REPLACEME-audit-log-dataset' with the correct project_id and dataset. You can also replace the `default-dataset` with your preferred dataset.
+
+### gcf_ingest_latest_by_table
+This view gives you the latest status for the latest chunk for a given table. If we wanted to see the latest chunk status for table 'table_a' we would run:
```
```suggestion
This view gives you the latest status for the latest load / query job for a given table. If we wanted to see the latest chunk status for table 'table_a' we would run:
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_latest_by_table.sql:
```diff
@@ -0,0 +1,33 @@
+CREATE OR REPLACE VIEW
```
Add Apache 2.0 header / google copyright
```suggestion
/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

CREATE OR REPLACE VIEW
```

jaketf replied:
add a comment that describes the purpose of the view.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_latest_by_table.sql:
```diff
@@ -0,0 +1,33 @@
+CREATE OR REPLACE VIEW
+  `default-dataset`.gcf_ingest_latest_by_table AS
```
Let's use a better name than default-dataset.
I don't think there can be `-` in dataset names FYI https://cloud.google.com/bigquery/docs/datasets#dataset-naming
Perhaps `gcf_ingest_monitoring_views`

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_latest_by_table.sql:
```diff
@@ -0,0 +1,33 @@
+ORDER BY
+  1,
+  2,
+  3;
```
nit: add a newline at end of all files.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_log.sql:
```diff
@@ -0,0 +1,44 @@
+  target_dataset_name as dataset_id,
+  target_table_name as table_id,
+  job_type,
+  -- Change this to extract the date and the chunk seperately (make this a UDF)
```
remove this comment I think what you're doing is generally fine.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_log.sql:
```diff
@@ -0,0 +1,44 @@
+  error_result
+FROM
+  region-us.INFORMATION_SCHEMA.JOBS_BY_PROJECT info_schema
+LEFT JOIN
```
I think you mean right join here to get just the tables that appear in monitored tables.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_log.sql:
```diff
@@ -0,0 +1,44 @@
+  destination_table.project_id,
+  destination_table.dataset_id,
+  destination_table.table_id,
+  start_time;
```
nit: newline

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/gcf_ingest_log.sql:
```diff
@@ -0,0 +1,44 @@
+  AND monitored_tables.target_dataset_name = info_schema.destination_table.dataset_id
+LEFT JOIN
+   `REPLACEME-audit-log-project-id.REPLACEME-audit-log-dataset.cloudaudit_googleapis_com_data_access` audit_log_sink 
+     ON info_schema.job_id = audit_log_sink.protopayload_auditlog.servicedata_v1_bigquery.jobCompletedEvent.job.jobName.jobId
```
what columns are coming from the audit logs? perhaps we don't need this join at all (and can remove all the complexity of REPLACEME*)

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/monitored_tables.sql:
```diff
@@ -0,0 +1,14 @@
+CREATE OR REPLACE TABLE
```
Add apache 2.0 header comment

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/monitored_tables.sql:
```diff
@@ -0,0 +1,14 @@
+  `default-dataset`.monitored_tables
+  (
+
+    table_group              STRING
```
Add descriptions. It's not obvious what table_group or load_type mean.

jaketf reviewed (changes requested):
I'd like if the SQL had more structured place holders for dataset / project ids and a script to render them.
For example you could use [envsubst](https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html)
This way we could render them and dry run them in the cloud build so we'd have some basic level of CI on these queries.

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md

Re-Wording

Co-authored-by: Jacob Ferriero <jferriero@google.com>

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md

Co-authored-by: Jacob Ferriero <jferriero@google.com>

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md

Co-authored-by: Jacob Ferriero <jferriero@google.com>

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md

Co-authored-by: Jacob Ferriero <jferriero@google.com>

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md

Co-authored-by: Jacob Ferriero <jferriero@google.com>

Commit:
Merge branch 'sequencing-monitoring' of https://github.com/jaketf/bigquery-utils into sequencing-monitoring

RyandenOtter committed:
Refinement of the Views.
Adding additional information to the READMEmd

RyandenOtter committed:
Fix Cloudbuild

RyandenOtter committed:
Fix Cloudbuild

RyandenOtter committed:
Fix Cloudbuild

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/cloudbuild.yaml:
```diff
@@ -34,6 +34,13 @@ steps:
     - '--cache=true'
     - '--context=dir:///workspace/${_BUILD_DIR}'
   id: 'build-ci-image'
+- name: 'gcr.io/$PROJECT_ID/gcs_event_based_ingest_ci'
```
last build failed because google cloud sdk not installed in this image.
just use [google/cloud-sdk](https://hub.docker.com/r/google/cloud-sdk/)
```suggestion
- name: 'google/cloud-sdk'
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/gcs_ocn_bq_ingest/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,42 @@
+}
+
+echo "setting DATASET env variable"
+export DATASET='test_monitoring_dataset'
```
there may be parallel tests running.
dataset id should be unique.
I suggest suffix with BUILD_ID which is [set by GCB automatically](https://cloud.google.com/cloud-build/docs/configuring-builds/substitute-variable-values).

```suggestion
export DATASET="test_monitoring_dataset_${BUILD_ID}"
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
````diff
@@ -60,12 +60,91 @@ DEFAULT_DESTINATION_REGEX = (
-`
+```
+
+## Monitoring
````
can you please move this info to the README in parent dir (e.g. replace the monitoring section in that README).

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/monitoring_tables_and_views.sql:
```diff
@@ -0,0 +1,105 @@
+/*
```
please move this monitoring directory to gcs_event_based_ingest.
otherwise we will package these files into the cloud function on each deploy (for no reason as the cloud function does not use these files).
Rather than adding `.gcloudignore` it would be cleaner to have this live in the gcs_event_based_ingest/monitoring level because this has no relation to the python source code in the gcs_ocn_bq_ingest module

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/monitoring_tables_and_views.sql:
```diff
@@ -0,0 +1,105 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+CREATE OR REPLACE TABLE
```
If you re-run this script it might drop data in this table.
```suggestion
CREATE TABLE IF NOT EXISTS
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/README.md:
````diff
@@ -60,12 +60,91 @@ DEFAULT_DESTINATION_REGEX = (
+```
+
+## Monitoring
+### `monitored_tables`
````
Add a note on how to deploy the monitoring scripts (e.g. snippet of setting DATASET env var and run your shell script)

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/monitoring_tables_and_views.sql:
```diff
@@ -0,0 +1,105 @@
+    target_table_name        STRING
+  );
+
+CREATE OR REPLACE VIEW
```
Let's have separate SQL file per view (shell script loops over files in a directory anyway)

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/monitoring/monitoring_tables_and_views.sql:
```diff
@@ -0,0 +1,105 @@
+ORDER BY
+  1,
+  2,
+  3;
```
nit: newline at end of file
[context](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/gcs_ocn_bq_ingest/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,42 @@
+  fi
+done <  <(find gcs_ocn_bq_ingest/monitoring -path "*.sql")
+echo "removing dataset $DATASET"
+bq rm -r -f -d $DATASET
```
nit: newline at end of file

jaketf replied:
note you have `set -e` which will fail and exit after first failed statement. [reference](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html)

This means if a query fails you will not clean up this dataset.
(see suggestion above using [trap](https://www.linuxjournal.com/content/bash-trap-command) and remove here)

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/gcs_ocn_bq_ingest/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,42 @@
+
+echo "setting DATASET env variable"
+export DATASET='test_monitoring_dataset'
+bq --location=US mk -f -d $DATASET
```
```suggestion
bq --location=US mk -f -d "$DATASET"
trap "bq rm -r -f -d $DATASET" EXIT
```

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/cloudbuild.yaml

Co-authored-by: Jacob Ferriero <jferriero@google.com>

RyandenOtter committed:
Update tools/cloud_functions/gcs_event_based_ingest/tests/gcs_ocn_bq_ingest/test_monitoring_tables.sh

Co-authored-by: Jacob Ferriero <jferriero@google.com>

RyandenOtter committed:
Monitoring files:
Move the monitoring.sql files up a directory
Split out the monitoring sql into its own files (Note the numbering is to ensure the files are listed in the correct order because they are all dependent on one another)

test_monitoring_tables.sh
Made some alterations to test_monitoring_tables.sh to ensure dataset gets cleaned up even if the sqls error.
removed the dry run argument from the bq query to ensure the sql is actually applied.

Readme:
Moved the Monitoring section out to the higher-level README

cloudbuild.yaml
Updated the cloudbuild to point to the new location of the tests.

RyandenOtter committed:
Add 'IF NOT EXISTS' to sql CREATE statements.

RyandenOtter committed:
Add instructions on deploying the files. [skip ci]

RyandenOtter committed:
Add more information about what each table/view does.

Fix 03_gcf_ingest_latest_by_table.sql

Some more README fixes.

RyandenOtter commented:
Hey! It looks like I don't have permissions to delete datasets in your test project, which is causing my local tests to fail.
It also looks like the gcs_event_based_ingest_ci image doesn't have envsubst installed, which is causing the build to fail.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/monitoring/01_monitored_tables.sql:
```diff
@@ -0,0 +1,28 @@
+/*
```
is there a reasoning for the numbering 01, 02, 03 in these query filenames? can we remove that?

jaketf replied:
I see you need to run these sql in a certain order because they create metadata entries that you depend on existing and you're doing this with `sort`. If there's a way to just have an ordered list in the shell script that might be better than confusing filenames IMO.

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/monitoring/02_gcf_ingest_log.sql:
```diff
@@ -0,0 +1,57 @@
+  destination_table.project_id,
+  destination_table.dataset_id,
+  destination_table.table_id,
+  start_time;
```
newline

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/monitoring/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,41 @@
+  fi
+done <  <(find monitoring -path "*.sql" | sort)
+echo "removing dataset $DATASET"
+trap "bq rm -r -f -d $DATASET" EXIT
```
nit. this line will never get executed if there is a failure above. Trap statement should be moved to like 29 (see suggestion above).
```suggestion
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/monitoring/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,41 @@
+    exit "$result"
+  fi
+done <  <(find monitoring -path "*.sql" | sort)
+echo "removing dataset $DATASET"
```
```suggestion
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/monitoring/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,41 @@
+
+export DATASET="test_monitoring_dataset_${BUILD_ID}"
+echo "Creating ${DATASET}"
+bq --location=US mk -f -d "$DATASET"
```
```suggestion
bq --location=US mk -f -d "$DATASET"
trap 'echo "removing dataset $DATASET" && bq rm -r -f -d $DATASET' EXIT
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/tests/monitoring/test_monitoring_tables.sh:
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Copyright 2019 Google Inc.
```
```suggestion
# Copyright 2021 Google Inc.
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/monitoring/03_gcf_ingest_latest_by_table.sql:
```diff
@@ -0,0 +1,51 @@
+/*
+# Copyright 2020 Google LLC
```
```suggestion
# Copyright 2021 Google LLC
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/monitoring/02_gcf_ingest_log.sql:
```diff
@@ -0,0 +1,57 @@
+/*
+# Copyright 2020 Google LLC
```
```suggestion
# Copyright 2021 Google LLC
```

jaketf commented on tools/cloud_functions/gcs_event_based_ingest/monitoring/01_monitored_tables.sql:
```diff
@@ -0,0 +1,28 @@
+/*
+# Copyright 2020 Google LLC
```
```suggestion
# Copyright 2021 Google LLC
```

jaketf commented:
You can just install envsubst in Dockerfile.ci
